File: TwoSum/TwoSum.py
#Beats 98.3% of Python3 solution for Memory Usage  
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        compliments = []
        for idx, val in enumerate(nums):
            if val in compliments:
                return [compliments.index(val), idx]
            else:
                compliments.append(target-val)
                
#Average solution, nothing really special, but 10x faster than the memory-efficiency solution above.                
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        compliments = dict()
        for idx, val in enumerate(nums):
            if compliments.get(val, None) != None:
                return [compliments.get(val), idx]
            else:
                compliments[target-val] = idx

File: TwoSum/TwoSum.cpp
// 99.79% faster than C++ solutions
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> compliments;
        for (int i = 0; i < nums.size(); i++) {
            if (compliments.find(target - nums[i]) != compliments.end())
                return vector<int> {compliments[target - nums[i]], i};
            else
                compliments[nums[i]] = i;
        }
        return vector<int> {-1, -1};
    }
};